sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -1483,7 +1491,7 @@ def test_enet_alpha_max_sample_weight(X_is_sparse, fit_intercept, sample_weight)
     assert_allclose(reg.coef_, 0, atol=1e-5)
     alpha_max = reg.alpha_
     # Test smaller alpha makes coefs nonzero.
-    reg = ElasticNet(alpha=0.99 * alpha_max, fit_intercept=fit_intercept)
+    reg = ElasticNet(alpha=0.95 * alpha_max, fit_intercept=fit_intercept)
```

Comment:
Have you checked that the difference in convergence criterion value in `main` and this branch with `alpha=0.99` still lies within `tol`?

From a cursory glance, I have the feeling that this test has become a bit weaker and this might reveal a potential bug.